### ISSUE TITLE
iOS/Android: Allowance of focus on non-view-backed elements

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30853]
## Requires API Proposal
This issue requires a new API. An API proposal should be added and discussed before proceeding with implementation. The API proposal can be added in the comments of this issue or linked as a separate issue.

## Description

The ability to create "virtual views" for accessibility would allow for making significantly more complex UI (charts, graphs, etc.), or have significant performance improvements (rendering multiple images to one bitmap, while keeping them individually focusable).

An API would need to be created that allows you to mark a component as not needing to create an actual View, but instead a virtual view for accessibility. This would likely be used to create virtual child views of some actual view-backed parent. These virtual views would not have any impact for users who aren't using accessibility services, and could simply not be rendered at all for these users.

## React Native version:
v0.63

## Expected Behavior
Components marked as virtual views would not create actual native views when rendered, but instead create virtual views for accessibility only.

## Android Details

On Android, virtual views can be created by using an AccessibilityDelegate, such as AndroidX's included ExploreByTouchHelper class. This is not a very straightforward process and not very well documented from Google. It  would require some deep diving into the system that translates JS components into native Views.

## iOS Details

On iOS, this is also significantly complex, and would require the creation of UIAccessibilityElement's directly, and setting them as the array of accessibilityElements from some parent UIAccessibilityContainer.
